packages/flutter/test/widgets/selectable_region_test.dart

```diff
@@ -3594,6 +3594,25 @@ void main() {
     skip: kIsWeb, // [intended] Web uses its native context menu.
   );
 
+  // Regression test for https://github.com/flutter/flutter/issues/121053.
+  testWidgets('Ensure consistent layout of SelectionArea children between platforms', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      const MaterialApp(
+        home: Column(
+          crossAxisAlignment: CrossAxisAlignment.stretch,
+          children: <Widget>[
+              SelectionArea(child: Text('row 1')),
+              Text('row 2'),
+            ],
+        ),
+      ),
+    );
+    await tester.pumpAndSettle();
+    final double xOffset1 = tester.getTopLeft(find.text('row 1')).dx;
+    final double xOffset2 = tester.getTopLeft(find.text('row 2')).dx;
+    expect(xOffset1, xOffset2);
+  });
```

Comment:
I think we should probably run this test on all platforms with `variant: TargetPlatformVariant.all()`.